_posts/2016-05-26-Rust-1.9.md

```diff
@@ -0,0 +1,155 @@
+
+As always, you can [install Rust 1.9][install] from the appropriate page on our
+website, and check out the [detailed release notes for 1.9][notes] on GitHub.
+About 1400 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-190-2016-05-26
+
+### What's in 1.9 stable
+
+Rust 1.9 contains some exciting new things!
+
+First up, a new attribute for library authors: `#[deprecated]`. Authors of
+libraries can now tag items with this attribute, and users of their crate
+will see a warning when it is used. This new feature was defined in [RFC 1270].
+
+[RFC 1270]: https://github.com/rust-lang/rfcs/blob/master/text/1270-deprecation.md
+
+We now publishes std binaries for the `mips-unknown-linux-musl`,
+`mipsel-unknown-linux-musl`, and `i586-pc-windows-msvc` targets.
+
+[The time complexity of comparing variables for equivalence][compare] during
+type unification is reduced from O(n!) to O(n). What does this mean? Certain
+kinds of code compiles much, much faster.
+
+[compare]: https://github.com/rust-lang/rust/pull/32062
+
+Strings are a very common type, and implement a lot of different traits.
+Specifically, when converting between `&str` and `String`, you have many
+options. Rustaceans used these two the most:
+
+```rust
+let s = "some string".to_string();
+let s = "some string".to_owned();
+```
+
+The former is the official convention, but it was discovered that it's not
+as fast as the second method. For a long time, it was not possible to improve
+this situation, however, a new language feature has landed on nightly and made
+this possible. [`libstd` now uses this internally][str] to make the two equivalent.
+
+[str]: https://github.com/rust-lang/rust/pull/32586
+
+Finally, this feature is technically a library change, but it's worth
+mentioning here as well. The `std::panic` module is being stabilized. This
+allows panics to be caught:
+
+```rust
+use std::panic;
+
+let result = panic::catch_unwind(|| {
+    println!("hello!");
+});
+assert!(result.is_ok());
+
+let result = panic::catch_unwind(|| {
+    panic!("oh no!");
+});
+assert!(result.is_err());
+```
+
+This behavior was defined in [RFC 1236].
+
+[`std::panic`]: http://doc.rust-lang.org/stable/std/panic/index.html
+[RFC 1236]: https://github.com/rust-lang/rfcs/pull/1236
+
+Historically, this has not been possible in Rust: panics are always
+unrecoverable. And in practice, this is still how they should be used. While
+the current implementation of `panic!` does unwinding, in a future release,
+crates will be able to choose a different implementation, namely, that
+`panic!` causes an abort, rather than unwind. Given that panics are for
+non-recoverable errors only, this is a valid implementation.
+
+If panics are for non-recoverable errors, why provide an interface for
+recovering from them? There are two big use-cases here:
```

Comment:
Yeah I might reorganize these sections as well and mention the FFI use case first and then just mention later that panics are still intended for "unrecoverable" errors